### PR TITLE
Publicize: point wp-admin users to WP.com calypso sharing page instead

### PR DESCRIPTION
**Pull request by jkudish**
This PR seeks to remove the publicize connection settings from WP.com's wp-admin and instead point users to the Calypso sharing page for their site.

screenshot from 2015-12-18 14-55-01

The code that powers the wp-admin sharing screen is considered legacy code, and is out of date with our latest Publicize connections codebase (powered by the WP.com REST API). We're taking aim at cleaning up old, less reliable code and reducing our technical debt, while offering the best user experience for users.

The XML-RPC methods will continue to be supported for the foreseeable future, even though the UI will be gone, but may be retired at some point once usage of older versions of Jetpack drops to low enough numbers.

With that in mind, it would be possible to re-create a wp-admin experience using the REST API. All of the endpoints are fully available and documented here: https://developer.wordpress.com/docs/api/#sharing

This PR is a merge of r128505-wpcom with additional removals for some JP-specific code.

To test:

* Apply PR to your Jetpack site
* Activate Publicize
* Verify that you get the screenshot above in place of the publicize settings
* Verify that the button and link take you to the expected Calypso Sharing page
* Verify that you can still add and remove publicize connections
* Verify that your added connections show up in wp-admin's Editor and Calypso Editor's as available connections
* Verify that posting a post publicizes the post as expected
* Verify that all of the above occurs without any php errors in your site's error log
* Login with a secondary user on your JP site, verify that you're still presented with this:

screenshot from 2015-12-18 14-54-46
 

cc @timmyc @aduth @nylen for review

### CODE REVIEW

File: modules/publicize/ui.php [hunk 1]
```diff
@@ -52,234 +51,23 @@ function management_page() { ?>
...
-			Publicize_UI::connected_notice( $service_name );
-			$update_notice = ob_get_clean();
-			if ( ! empty( $update_notice ) )
-				echo $update_notice;
+		<h3 id="publicize"><?php esc_html_e( 'Publicize', 'jetpack' ) ?></h3>
+		<p><?php esc_html_e( 'Connect social media services to automatically share new posts.', 'jetpack' ) ?></p>
+		<h4><?php
+			printf(
+				wp_kses(
+					__( "We've recently made some updates to Publicize. Please visit the <a href='%s'>WordPress.com sharing page</a> to manage your publicize connections or use the button below.", 'jetpack' ),
```
Comment (kraftbj) [inline modules/publicize/ui.php]: I'm afraid "recently" will become outdated pretty soon. Is there a better phrase or can we be sure to update it in, say, six months to something else?

Comment (jkudish) [inline modules/publicize/ui.php]: Eventually we can aim to remove this sentence all together, but yeah I see your point.

I don't want it to be too matter of fact without providing some context that this is an expected "new-ish" change.

Do you have a suggestion?

Comment (aduth) [inline modules/publicize/ui.php]: Could we just drop "recently" ? Still conveys improvements, but without as much of a time implication.

Comment (nylen) [inline modules/publicize/ui.php]: +1 to dropping "recently".  Also, "publicize" should be capitalized in "to manage your publicize connections".

**Comment by dereksmart** [comment]
this is pretty awesome :+1:

**Comment by dereksmart** [comment]
actually @jkudish I can't get posts to actually publicize with this patch.

**Comment by jkudish** [comment]
> edit: might be my local setup. Will look more tomorrow.

That's likely. This PR doesn't affect the actual publicizing of posts.

**Comment by dereksmart** [comment]
Yeah confirmed it's something broken with my site & publicize.  Won't publicize from any branch.  Others have confirmed this works. 

:shipit:

**Pull request merged**